src/borg/testsuite/chunker.py

```diff
@@ -59,3 +60,30 @@ def read(self, nbytes):
         chunker = get_chunker(*CHUNKER_PARAMS, seed=0)
         reconstructed = b''.join(chunker.chunkify(SmallReadFile()))
         assert reconstructed == b'a' * 20
+
+    def test_random_input(self):
+        def twist(size):
+            x = 1
+            a = bytearray(size)
+            for i in range(size):
+                x = (x * 1103515245 + 12345) & 0x7FFFFFFF
+                a[i] = x & 0xFF
+            return a
+
+        data = twist(100000)
+
+        h = sha256()
+        for winsize in (65, 129, HASH_WINDOW_SIZE, 7351):
+            for minexp in (4, 6, 7, 11, 12, CHUNK_MIN_EXP):
+                for maxexp in (15, 17, CHUNK_MAX_EXP):
+                    if minexp >= maxexp:
+                        continue
+                    for maskbits in (4, 7, 10, 12, HASH_MASK_BITS):
+                        for seed in (1849058162, 1234567653):
+                            fh = BytesIO(data)
+                            chunker = Chunker(seed, minexp, maxexp, maskbits, winsize)
+                            for c in chunker.chunkify(fh, -1):
+                                h.update(sha256(c).hexdigest().encode())
+                            h.update(b"\n")
+
+        self.assert_equal(h.hexdigest(), "ef6ba5fbf60391af0b45692c991fa2490b0fd5865afb337f52498a677a2bd670")
```

Comment:
there should be a comment above this line about how this hash was computed.

Reply:
adding one